clippy_lints/src/methods/mod.rs

```diff
@@ -3826,6 +3866,24 @@ fn lint_filetype_is_file(cx: &LateContext<'_>, expr: &hir::Expr<'_>, args: &[hir
     span_lint_and_help(cx, FILETYPE_IS_FILE, span, &lint_msg, None, &help_msg);
 }
 
+fn lint_from_iter(cx: &LateContext<'_>, expr: &hir::Expr<'_>, args: &[hir::Expr<'_>]) {
+    let ty = cx.typeck_results().expr_ty(expr);
+    let id = get_trait_def_id(cx, &paths::FROM_ITERATOR_TRAIT).unwrap();
+
+    if implements_trait(cx, ty, id, &[]) {
```

Comment:
You should be able to just use
```suggestion
    if match_trait_method(cx, expr, &paths::FROM_ITERATOR) {
```

here.

Reply:
Initially I tried this solution but it panics every time.
Message: `thread 'rustc' panicked at 'called `Option::unwrap()` on a `None` value', clippy_lints/src/utils/mod.rs:164:73`

Reply:
Can you try to use the full path to `FromIterator` instead: `["core", "iter", "traits", "collect", "IntoIterator"]`? The `std::iter::FromIterator` path is only a reexport, which may cause the panic.

Reply:
Still panics.

Reply:
Oh maybe because `from_iter` is a function, not a method . Alright, can you change the `FROM_ITERATOR` path to `["core", "iter", "traits", "collect", "FromIterator"]` either way, for consistency with the other `ITERATOR` paths?